presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst

```diff
@@ -0,0 +1,97 @@
+===========================
+History based optimizations
+===========================
+
+What is HBO
+------------
+
+HBO is a framework which enables recording statistics of a query to reuse the statistics for future queries with similar plans. To match queries with similar plans, a query plan
+is canonicalized so as to get rid of unrelated differences between plans (for example naming of intermediate variables in a plan), and each plan node is hashed to a string value.
+Historical statistics associated with the same hash value are used in HBO. HBO statistics is preferred over estimations from cost model, and optimizer falls back to cost based
+statistics when history statistics are not available.
+The statistics recorded for HBO are in `PlanStatistics` structs, which includes:
+
+=============================== =============================================================================================================== ===========================================================================================================
+Field                           Description                                                                                                     Optimizers
+=============================== =============================================================================================================== ===========================================================================================================
+rowCount                        Number of rows output by a plan node                                                                            DetermineJoinDistributionType, DetermineSemiJoinDistributionType, PushPartialAggregationThroughExchange
```

Comment:
You could remove the commas after the three optimizers here and in the following line - the formatting of the three names separates them well enough.